app/models/metric/purging.rb

```diff
@@ -85,9 +87,32 @@ def self.purge_hourly(older_than, window = nil, total_limit = nil, &block)
   end
 
   def self.purge_realtime(older_than, window = nil, total_limit = nil, &block)
-    purge_by_date(older_than, "realtime", window, total_limit, &block)
+    truncate_child_tables(older_than)
   end
 
+  def self.truncate_child_tables(older_than)
```

Comment:
This and the below method should really be documented I think.  There is a decent amount of domain knowledge that comes with them to understand how they work fully (for example, the mod division logic you do to figure out the good hours for dates that might be between two different days).  I would also be surprised if most developers even knew that the metrics table was split into 24 parts on hour (again, I didn't until today... and only new that it was sharded).